weather-card/weather-card.js

```diff
@@ -144,84 +144,95 @@ class WeatherCard extends LitElement {
     return html`
       ${this.renderStyle()}
       <ha-card @click="${this._handleClick}">
-        <span
-          class="icon bigger"
-          style="background: none, url(${
-            this.getWeatherIcon(
-              stateObj.state.toLowerCase(),
-              this.hass.states["sun.sun"].state
-            )
-          }) no-repeat; background-size: contain;"
-          >${stateObj.state}
-        </span>
         ${
-          this._config.name
-            ? html`
-                <span class="title"> ${this._config.name} </span>
+          this._config.compact
+            ? ""
+            : html`
+                <span
+                  class="icon bigger"
+                  style="background: none, url(${
+                    this.getWeatherIcon(
+                      stateObj.state.toLowerCase(),
+                      this.hass.states["sun.sun"].state
+                    )
+                  }) no-repeat; background-size: contain;"
+                  >${stateObj.state}
+                </span>
+                ${
+                  this._config.name
+                    ? html`
+                        <span class="title"> ${this._config.name} </span>
+                      `
+                    : ""
+                }
+                <span class="temp"
```

Comment:
No functional changes here, looks like my formatter just went to town on this section. I can revert this if you like